src/App.jsx

```diff
@@ -14,45 +16,138 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABETICALLY = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_REVERSE = 'reverse';
+
+function getPreparedGood(goods, sortField, revers) {
+  const visibleGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_BY_ALPHABETICALLY:
+        if (revers) {
+          visibleGoods.sort((good1, good2) => good2.localeCompare(good1));
+        } else {
+          visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+        }
+        break;
+      case SORT_BY_LENGTH:
+        if (revers) {
+          visibleGoods.sort((good1, good2) => (
+            good2.length - good1.length))
+        } else {
+          visibleGoods.sort((good1, good2) => (
+            good1.length - good2.length))
+        }
+        break;
+      case SORT_BY_REVERSE:
+        visibleGoods.reverse();
+        break;
+      default:
+        break;
+    }
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(null);
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [reverse, setReverse] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
It's a naming convention for boolean states